app/audit/models/YesPreferenceAttemptedAuditModel.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import play.api.libs.json._
+
+case class YesPreferenceAttemptedAuditModel(agentReferenceNumber: String,
+                                            notificationPreference: String = "Notification required",
+                                            attemptedNotificationEmailAddress: String) extends ExtendedAuditModel {
+
+  override val transactionName: String = YesPreferenceAttemptedAuditModel.agentTransactionName
+  override val detail: JsValue = Json.toJson(this)
+  override val auditType: String = YesPreferenceAttemptedAuditModel.agentAuditType
+}
+
+object YesPreferenceAttemptedAuditModel {
```

Comment:
Is this helper object needed? It seems unnecessary to have the transaction name and audit type defined below and have the class call on them, rather than just define them itself. Unless I'm missing something.